Consolidate component explanation, including classic and path ids, hierarchy, addressing

emiltin opened the pull request:
components.rst now described fundamental concepts:

 - what a component is
- type, id, name, data
- id formats: classic and path
- how to address components

for now it leaves out any reference to component list, and also the section on component order/index (since this relies on the component list). also leaves out reference to sxl modules and hiererchy.

emiltin committed:
update components.rst

emiltin committed:
describe component data

FTjneuvonen reviewed (approved):
OK.

otterdahl commented:
A few comments:

1. It defines "component", "component type" and "component id", but they are already defined in `definitions.rst`. I don't think we should define them again. I think we should decide on a single location.

2. "Legacy Id". I think we should stick to just calling them "type A" and "type B"? I'm unsure if we (STA) may ever move away from our existing id's (but you never know).

There's also a number of formatting issues.

3. There needs to be a blank line between explicit references and titles:
```
.. _components:

Components
==========

```
or else the causes the following error message: ` WARNING: Explicit markup ends without a blank line; unexpected unindent.`

4. When you change name of a document you need to update the index at `source/applicability.rst` or else it won't be included.
```
applicability/components.rst
```

5. There is an invalid ":ref:`Site id`". It should be ":term:`Site id`"

6. Spelling error, "componets"

There might be more formatting fixes needed, but we need to fix these first...

emiltin committed:
components.rst fixes

emiltin commented:
> 1. It defines "component", "component type" and "component id", but they are already defined in `definitions.rst`. I don't think we should define them again. I think we should decide on a single location.

I think we need to explain things in context, with examples, etc. So I would prefer to move the main parts to components.rst. The definitions.rst can be an index of concepts, some can be briefly explained, others can link to relevant sections?


> 2. "Legacy Id". I think we should stick to just calling them "type A" and "type B"? I'm unsure if we (STA) may ever move away from our existing id's (but you never know).

OK so let's not use "legacy". I still think it would be great if we can find a descriptive name.  What about 'original' or 'fixed'? I found 'flat' to be intuitive, since there are no hierarchy in the component parts, as opposed to the path format. I feel 'nordic' goes a bit against the ideas of being country/region neutral.

> 3. There needs to be a blank line between explicit references and titles:

fixed

> 4. When you change name of a document you need to update the index at `source/applicability.rst` or else it won't be included.

fixed

>  5. There is an invalid ":ref:`Site id`". It should be ":term:`Site id`"

fixed. I will try to get the build system installed so I can build the docs, so I can catch these kinds of errors. Btw, maybe we could set that up as a github action workflow that runs on PR pushes?

> 6. Spelling error, "componets"

fixed

> There might be more formatting fixes needed, but we need to fix these first...

I fixed a few other typos.

otterdahl commented:
> > 1. It defines "component", "component type" and "component id", but they are already defined in `definitions.rst`. I don't think we should define them again. I think we should decide on a single location.
> 
> I think we need to explain things in context, with examples, etc. So I would prefer to move the main parts to components.rst. The definitions.rst can be an index of concepts, some can be briefly explained, others can link to relevant sections?

Sure. That makes sense. Let's work on a compromise.

> OK so let's not use "legacy". I still think it would be great if we can find a describe name.  What about 'original' or 'fixed'? I would 'flat' to be intuitive, since there are no hierarchy in the component parts, as opposed to the path format. I feel 'nordic' goes a bit against the ideas of being country/region neutral.

Sure. 'original' (or 'classic'?) are ok alternatives to "type A". I think 'flat' is a bit confusing, since in our road authority we consider them to be hierarchical.

> I will try to get the build system installed so I can build the docs, so I can catch these kinds of errors. Btw, maybe we could set that up as a github action workflow that runs on PR pushes?

Yeah. I can open up a new PR for that.

emiltin committed:
classic component id format, trim definitions.rst

emiltin committed:
fix :refs

emiltin commented:
I addressed your comments.
Thanks for adding the build action. I see that warnings do not cause the action to fail. i got the following warning, but the action still succeeded:

/home/runner/work/rsmp_core/rsmp_core/source/definitions.rst:81: WARNING: undefined label: 'component'

Maybe we should be stricter?

emiltin commented:
Do we want to allowed component subtypes? Example the SXL ```sensor``` might define:

```
dt
dt/video
dt/radar
```

I lean towards leaving that out for now, although it could be purely naming, with no functional impact.

otterdahl commented:
> Do we want to allowed component subtypes? Example the SXL ```sensor``` might define:
> 
> ```
> dt
> dt/video
> dt/radar
> ```

Could we image any specific use case? I see no use case right now. I suggest leaving it out for now.

emiltin commented:
Not in the current TLC sxl. But if we define an SXL for sensor, component types for e.g radar, camera etc might make sense. Some status relate to all types (counts) while specialized functionality could be defined for e.g. a camera (image snapshot, video link..?)

But we can leave it out for now.

otterdahl committed:
Merge branch '3.3.0' into components_concepts

otterdahl committed:
workflow: Exit with error on any warning

otterdahl commented:
> Thanks for adding build action. I see that warnings do not cause the test to fail. Maybe we should be stricter?

Agree. Should be fixed now

otterdahl committed:
Revert "workflow: Exit with error on any warning"

This reverts commit b64ffaa715d2515ced3013958ae79f71e887ab35.

otterdahl committed:
workflow: Exit with error on any warning (2)

emiltin committed:
fix :ref

emiltin commented:
@otterdahl i fixed the ref error. do you think there are more conceptual work to do, or should we merge this to 3.3.0?

emiltin committed:
remote '/' from examples of component path ids

emiltin commented:
I removed "/" from examples of component path ids, as it's not a component id, but the root path.

emiltin committed:
use AB+84001 as classic id example

emiltin commented:
I noticed that the example using KK do not not match the classic format. In fact it seems the ids we use in Copenhagen generally does not match, e.g KK+AG2804=001SG008. Notice that "AG2804" has six chars, not five as the docs require. Compare:

```
AA+BBCDD=EEEFFGGG
AB+84001=860DL001
KK+AG2804=001SG008
```

I'll open a new issue for this, but for now I changed the example to AB+84001=860DL001.

otterdahl commented:
On Tue, Nov 25, 2025 at 01:39:37AM -0800, Emil Tin wrote:
> emiltin left a comment (rsmp-nordic/rsmp_core#258)
> 
> @otterdahl fix formatting errors. do you think there are more conceptual work to do, or should we merge this to 3.3.0?

I think it is good enough and ready for merge.

Just a final thought, please consider writing clear name for the PR. It makes it easier to write the changelog later on without doing detective work :-)

emiltin commented:
I updatd the PR description, what do you think?

otterdahl commented:
> I updatd the PR description, what do you think?

A lot better. Thank you!

emiltin commented on source/applicability/components.rst:
```diff
@@ -0,0 +1,141 @@
+
+Component types are used in SXLs when defining which alarms, commands and statuses apply to which components.
+
+Component types must be unique within the SXL where they are defined, but does not have to be globally unique.
```
this is a comment
